Dockerfile

```diff
@@ -1,53 +1,73 @@
 # Use the official Ruby image because the Rails images have been deprecated
-FROM ruby:2.5
+FROM ruby:2.5-slim
 
-# Enable https
-RUN apt-get update
-RUN apt-get install -y apt-transport-https
+# Enable package fetch over https and add a few core tools
+RUN apt-get update \
+    && apt-get install -y \
+       apt-transport-https \
+       curl \
+    && rm -rf /var/lib/apt/lists/*
 
 # Install Postgres client
-RUN apt-get install -y --no-install-recommends postgresql-client
-RUN rm -rf /var/lib/apt/lists/*
-
-# Install Chrome for capybara
-RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - 
-RUN sh -c 'echo "deb https://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list'
-RUN apt-get update
-RUN apt-get install -y google-chrome-stable
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+       postgresql-client \
+       libpq-dev \
+    && rm -rf /var/lib/apt/lists/*
 
 # Install Node 12.x
-RUN curl -sL https://deb.nodesource.com/setup_12.x | bash -
-RUN apt-get install -y nodejs
-RUN ln -s ../node/bin/node /usr/local/bin/
-RUN ln -s ../node/bin/npm /usr/local/bin/
+RUN curl -sL https://deb.nodesource.com/setup_12.x | bash - \
+    && apt-get update \
+    && apt-get install -y nodejs \
+    && rm -rf /var/lib/apt/lists/* \
+    && ln -s ../node/bin/node /usr/local/bin/ \
+    && ln -s ../node/bin/npm /usr/local/bin/
 
 # Install Yarn
-RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
-RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
-RUN apt-get update && apt-get install yarn
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+    && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
+    && apt-get update \
+    && apt-get install yarn \
+    && rm -rf /var/lib/apt/lists/*
 
 # Everything happens here from now on   
 WORKDIR /upaya
 
 # Simple Gem cache.  Success here creates a new layer in the image.
-COPY Gemfile .
-COPY Gemfile.lock .
-RUN gem install bundler --conservative
-RUN bundle install --without deploy production
+# Note - Installs build related debs then removes after use
+COPY Gemfile Gemfile.lock ./
+RUN apt-get update \
+    && apt-get install -y \
+       build-essential \
+       git \
+       liblzma-dev \
+       patch \
+       ruby-dev \
+    && gem install bundler --conservative \
+    && bundle install --without deploy production \
+    && apt-get remove -y \
```

Comment:
I think this is just installing dependencies for bundling like `build-essential`, then removing them after we're doing installing the Gems.  Is that right?

Reply:
Correct!  All those build tools (gcc, make, etc) add up, so this gets rid of them before sealing the layer.